official/mnist/mnist.py

```diff
@@ -220,48 +231,23 @@ def eval_input_fn():
 
 
 class MNISTArgParser(argparse.ArgumentParser):
+  """Argument parser for running MNIST model."""
+  def __init__(self, **kwargs):
+    super(MNISTArgParser, self).__init__(parents=[
+      parsers.BaseParser(**kwargs),
+      parsers.ImageModelParser()])
 
-  def __init__(self):
-    super(MNISTArgParser, self).__init__()
-
-    self.add_argument(
-        '--multi_gpu', action='store_true',
-        help='If set, run across all available GPUs.')
-    self.add_argument(
-        '--batch_size',
-        type=int,
-        default=100,
-        help='Number of images to process in a batch')
-    self.add_argument(
-        '--data_dir',
-        type=str,
-        default='/tmp/mnist_data',
-        help='Path to directory containing the MNIST dataset')
-    self.add_argument(
-        '--model_dir',
-        type=str,
-        default='/tmp/mnist_model',
-        help='The directory where the model will be stored.')
-    self.add_argument(
-        '--train_epochs',
-        type=int,
-        default=40,
-        help='Number of epochs to train.')
-    self.add_argument(
-        '--data_format',
-        type=str,
-        default=None,
-        choices=['channels_first', 'channels_last'],
-        help='A flag to override the data format used in the model. '
-        'channels_first provides a performance boost on GPU but is not always '
-        'compatible with CPU. If left unspecified, the data format will be '
-        'chosen automatically based on whether TensorFlow was built for CPU or '
-        'GPU.')
     self.add_argument(
         '--export_dir',
         type=str,
         help='The directory where the exported SavedModel will be stored.')
```

Comment:
If you prepend "[default: %(default)s] " to the help string argparse will populate it when -h is called. (I've been doing this even when the default is None.) Probably also a good idea to indicate in the help string what the effect of not setting is. (Is the save file just in the model_dir or is it not created?)

Reply:
Thanks for the suggestions, I've added to the help string